src/rtcModule/webrtc.cpp

```diff
@@ -853,6 +906,51 @@ void RtcModule::abortCallRetry(Id chatid)
     }
 }
 
+void RtcModule::refreshTurnServerIp()
+{
+    if (mIceServerProvider.empty())
+    {
+        return;
+    }
+
+    std::string url = mIceServerProvider[0]->url;
+    if (url.size())
+    {
+        size_t postInitialColon = url.find(":") + 1;
+        size_t postFinalColon = url.rfind(":");
+        std::string subString = url.substr(postInitialColon, postFinalColon - postInitialColon);
+        auto wptr = weakHandle();
+        int turnServerDnsRequestId = ++mTurnServerDnsRequest;
+        DnsRequest::getInstance()->wsResolveDNS(mKarereClient.websocketIO, subString.c_str(),
+                                                [wptr, this, subString, turnServerDnsRequestId]
+                                                (int statusDNS, const std::vector<std::string> &ipsv4, const std::vector<std::string> &ipsv6)
+        {
+            if (wptr.deleted() || turnServerDnsRequestId != mTurnServerDnsRequest)
+                return;
+
+            if (!mKarereClient.mDnsCache.hasRecord(TURNSERVER_SHARE))
```

Comment:
Not sure if those records should be already added at this point. The right place to add them to the cache, IMHO, are in the lambda of `fetchServers()`. At that point the URL/s have been received and could be added to the cache.

Reply:
I follow this approach to keep to avoid look in cache the correspondence between URL and `Shard number`